components/backup-stream/src/endpoint.rs

```diff
@@ -165,6 +165,8 @@ where
             ((config.num_threads + 1) / 2).max(1),
         );
         pool.spawn(op_loop);
+        let mut checkpoint_mgr = CheckpointManager::default();
+        pool.spawn(checkpoint_mgr.spawn_subscription_mgr());
```

Comment:
is it necessary to wait for `checkpoint_mgr.manager_handle` to be ready, before return the `Endpoint`?

Reply:
The handle would be set upon `spawn_subscription_mgr` has been invoked, and subsequent messages can be buffered in the channel.🤔

Reply:
I see the `checkpoint_mgr.manager_handle` will be set by `channel::tx`, while `checkpoint_mgr.manager_handle` will be got by `as_ref` or `as_mut` in some other places.
I wonder whether `as_ref` and `as_mut` are atomic, like
```
thread1              |    thread2
*i = 1                    |   if *i == 1 {}
```

Reply:
Don't worry about that, the `&mut` requirement would make sure there won't be two threads concurrently access this field without lock as long as we are writing safe rust.

Reply:
get